MacOS standalone installer PR build failing

Seems like something is making the MacOS build fail on PRs as for example https://github.com/spyder-ide/spyder/pull/19617 https://github.com/spyder-ide/spyder/actions/runs/3129391951/jobs/5078662547

Checking the logs at the end, after the packages recipes are all checked, you can see the following:

```
*** filtering dependencies ***
15400 total
48 filtered
201 orphaned
15352 remaining
Error: Process completed with exit code 1.
```

Not totally sure what could be happening but just in case pinging @mrclary

--- a/installers/macOS/req-build.txt
+++ b/installers/macOS/req-build.txt
@@ -1,3 +1,4 @@
 # For building standalone Mac app
 dmgbuild>=1.4.2
 py2app>=0.28
+sphinx==5.1.1  # See spyder-ide/spyder#19618 for details.

